plugins/node/instrumentation-fs/test/fsPromises.test.ts

```diff
@@ -0,0 +1,199 @@
+      plugin.disable();
+      memoryExporter.reset();
+      context.disable();
+    });
+
+    const promiseTest: TestCreator = (
+      name: FPMember,
+      args,
+      { error, result, resultAsError = null },
+      spans
+    ) => {
+      const rootSpanName = `${name} test span`;
+      it(`promises.${name} ${error ? 'error' : 'success'}`, async () => {
+        const rootSpan = tracer.startSpan(rootSpanName);
+
+        assert.strictEqual(memoryExporter.getFinishedSpans().length, 0);
+        await context
+          .with(trace.setSpan(context.active(), rootSpan), () => {
+            // eslint-disable-next-line node/no-unsupported-features/node-builtins
+            assert(
+              typeof fsPromises[name] === 'function',
+              `Expected fsPromises.${name} to be a function`
+            );
+            return Reflect.apply(fsPromises[name], fsPromises, args);
+          })
+          .then((actualResult: any) => {
+            if (error) {
+              assert.fail(`promises.${name} did not reject`);
+            } else {
+              assert.deepEqual(actualResult, result ?? resultAsError);
+            }
+          })
+          .catch((actualError: any) => {
+            assert(
+              actualError instanceof Error,
+              `Expected caugth error to be instance of Error. Got ${actualError}`
+            );
+            if (error) {
+              assert(
+                error.test(actualError?.message ?? ''),
+                `Expected "${actualError?.message}" to match ${error}`
+              );
+            } else {
+              actualError.message = `Did not expect promises.${name} to reject: ${actualError.message}`;
+              assert.fail(actualError);
+            }
+          });
+        rootSpan.end();
+        assertSpans(memoryExporter.getFinishedSpans(), [
+          ...spans.map((s: any) => {
+            const spanName = s.name.replace(/%NAME/, name);
+            const attributes = {
+              ...(s.attributes ?? {}),
+            };
+            attributes[TEST_ATTRIBUTE] = TEST_VALUE;
+            return {
+              ...s,
+              name: spanName,
+              attributes,
+            };
+          }),
+          { name: rootSpanName },
+        ]);
+      });
+    };
+
+    const selection: TestCase[] = tests.filter(
+      ([name, , , , options = {}]) =>
+        options.promise !== false && name !== ('exists' as FsFunction)
+    );
+
+    describe('Instrumentation enabled', () => {
+      selection.forEach(([name, args, result, spans]) => {
+        promiseTest(name as FPMember, args, result, spans);
+      });
+    });
+
+    describe('Instrumentation disabled', () => {
+      beforeEach(() => {
+        plugin.disable();
+      });
+
+      selection.forEach(([name, args, result]) => {
+        promiseTest(name as FPMember, args, result, []);
+      });
+    });
+  });
+}
+
+const assertSpans = (spans: ReadableSpan[], expected: any) => {
```

Comment:
This function is the same as in fs.test.ts file. maybe it is better to put both of them in a 'utils.ts' file.